config/routes.rb

```diff
@@ -2,6 +2,15 @@
 require 'sidekiq/cron/web'
 
 Rails.application.routes.draw do
+  Sidekiq::Web.use Rack::Auth::Basic do |username, password|
+    # Protect against timing attacks:
+    # - See https://codahale.com/a-lesson-in-timing-attacks/
+    # - See https://thisdata.com/blog/timing-attacks-against-string-comparison/
+    # - Use & (do not use &&) so that it doesn't short circuit.
+    # - Use digests to stop length information leaking (see also ActiveSupport::SecurityUtils.variable_size_secure_compare)
+    ActiveSupport::SecurityUtils.secure_compare(::Digest::SHA256.hexdigest(username), ::Digest::SHA256.hexdigest(ENV["SIDEKIQ_USERNAME"])) &
+      ActiveSupport::SecurityUtils.secure_compare(::Digest::SHA256.hexdigest(password), ::Digest::SHA256.hexdigest(ENV["SIDEKIQ_PASSWORD"]))
+  end if Rails.env.production?
```

Comment:
so this will apply to all of UAT, Dev and Production, the only environments this removes it from is test and local dev. :)

Reply:
For me, the above chunk of code is quite complicated, even though it's short - I will confess I didn't even know you could avoid a boolean short-circuit by using `&`, so thanks for showing me! - so _ideally_ we'd have tests to prove that it lets you in with the right password and doesn't with the wrong one. It'd also be great to be able to sanity check this logic in a dev environment. So I'd prefer to remove the production check, but it's not a strong preference.

Reply:
>  so ideally we'd have tests to prove that it lets you in with the right password and doesn't with the wrong one.
 I personally would not write a test for it just because it is standard sidekiq web ui behaviour, albeit with a slight rails active support twist, so is that gems responsibility.

Reply:
just FYI - this is copied and pasted from the sidekiq website - most of the time was me looking adn reading to understand why this approach was recommended. :)